### DISCUSSION TITLE
Libaries for compiling HASPone in Arduino IDE

### DISCUSSION THREAD

**Discussion by Grevlingkroken** [HASwitchPlate/HASPone#37] [Q&A] [answered]
I have some issues upgrading from the old release to HASPone on my device (NodeMCU). OTA-Upgrade process goes fine, but MQTT-settings are lost. When trying to update with correct MQTT-info, the ESP will not join wifi anymore despite that I retyped the wifi-pwd as well. Tried four times on two different machines, so I figure compiling in Arduino IDE is the best choice. Do you have a list of libraries/version I need for this to succeed ? I can compile the HASwitchPlate with no issues, but HASPone fails. I added ESP8266ping and EspSaveCrash, but probably not the correct versions as it always fails. Anyone? Thanks!

**Comment by ielbury** [comment]
This is my platformio.ini which might give you some clues ... the versions might not be right either?

```
[env:d1_mini]
platform = espressif8266
board = d1_mini
framework = arduino
board_build.ldscript = eagle.flash.4m.ld
lib_deps = 
	krzychb/EspSaveCrash@1.2.0
	https://github.com/tzapu/WiFiManager.git#master
	bblanchon/ArduinoJson@^6.15.0
	dancol90/ESP8266Ping@1.0

```

**Comment by aderusha** [comment, marked as answer]
One thing that might be tripping you up - the version of [WiFiManager](https://github.com/tzapu/WiFiManager/commits/master) available in the Arduino IDE is pretty old.  You'll need to clone a recent copy of the master branch in the library repo.

**Reply by Grevlingkroken** [reply]
Thanks for the info - updating WifiManager from 0.16 to 2.0.3-alpha did the trick! :-)
